src/yachtzee.ts

```diff
@@ -0,0 +1,831 @@
+            potion.potion
+          } @ price ${price} meat each`,
+          "blue"
+        );
+        if (!simOnly) {
+          acquire(nPotions, potion.potion, profit + price);
+          if (itemAmount(potion.potion) < 1) break;
+          if (isSong(effect) && !have(effect)) {
+            for (const song of getActiveSongs()) {
+              const slot = Mood.defaultOptions.songSlots.find((slot) => slot.includes(song));
+              if (!slot || slot.includes(effect)) {
+                cliExecute(`shrug ${song}`);
+              }
+            }
+          }
+          if (
+            !potion.use(Math.min(itemAmount(potion.potion), nPotions)) ||
+            tries >= 5 * Math.ceil(yachtzeeTurns / potion.effectDuration())
+          ) {
+            break;
+          }
+        } else break;
+      }
+      if (have(effect) || simOnly) {
+        for (const excluded of mutuallyExclusive.get(effect) ?? []) {
+          excludedEffects.add(excluded);
+        }
+      }
+    }
+  }
+
+  if (!simOnly) {
+    executeNextDietStep(true);
+    if (have($item`Platinum Yendorian Express Card`) && !get(`expressCardUsed`)) {
+      use(1, $item`Platinum Yendorian Express Card`);
+    }
+  }
+
+  // Uncle Greenspan's may be cost effective
+  if (!simOnly && !have($effect`Buy!  Sell!  Buy!  Sell!`)) {
+    const yachtzeeFactor = yachtzeeTurns * (yachtzeeTurns + 1);
+    const embezzlerFactor =
+      Math.min(100, expectedEmbezzlers + yachtzeeTurns) *
+      (Math.min(100, expectedEmbezzlers + yachtzeeTurns) + 1);
+    const greenspanValue =
+      (2000 * yachtzeeFactor +
+        (baseMeat + 750) * (embezzlerFactor - yachtzeeFactor) +
+        baseMeat * (10100 - embezzlerFactor)) /
+      100;
+    const price = garboValue($item`Uncle Greenspan's Bathroom Finance Guide`);
+    const profit = greenspanValue - price;
+    if (profit > 0) {
+      print(
+        `Expected to profit ${profit} meat from using 1 Uncle Greenspan's Bathroom Finance Guide @ price ${price} meat`,
+        "blue"
+      );
+      acquire(1, $item`Uncle Greenspan's Bathroom Finance Guide`, greenspanValue);
+      if (have($item`Uncle Greenspan's Bathroom Finance Guide`)) {
+        use(1, $item`Uncle Greenspan's Bathroom Finance Guide`);
+      }
+    }
+  }
+
+  return totalProfits;
+}
+
+function _yachtzeeChain(): void {
+  if (!have($item`fishy pipe`)) return;
+  // hard require fishy pipe to run this chain
+  else if (!have($familiar`Urchin Urchin`)) return;
+  // also hard require urchin urchin for now
+  else if (myLevel() <= 13 || !canInteract()) return;
+  // We definitely need to be able to eat sliders and drink pickle juice
+  else if (property.getBoolean("fishyPipeUsed") && !have($effect`Fishy`)) return;
+  // If we have used our fishy pipe and have no fishy turns left, we're probably done
+  else if (
+    !property.getBoolean("sleazeAirportAlways") &&
+    !property.getBoolean("_sleazeAirportToday")
+  ) {
+    return;
+  }
```

Comment:
```suggestion
  if (!have($item`fishy pipe`)) return;
  // hard require fishy pipe to run this chain
  if (!have($familiar`Urchin Urchin`)) return;
  // also hard require urchin urchin for now
   if (myLevel() <= 13 || !canInteract()) return;
  // We definitely need to be able to eat sliders and drink pickle juice
  if (get("fishyPipeUsed") && !have($effect`Fishy`)) return;
  // If we have used our fishy pipe and have no fishy turns left, we're probably done
  if (!realmAvailable("sleaze")) return;
```

You'll need to import `realmAvailable`, but that's fine and valid

Reply:
I also feel like if you can't interact we should just categorically not work right now, but that's another bridge to cross